rewrite-pattern.js

```diff
@@ -300,42 +307,84 @@ const processTerm = (item, regenerateOptions, groups) => {
 	return item;
 };
 
+// Enable every stable RegExp feature by default
+const regjsparserFeatures = {
+	'unicodePropertyEscape': true,
+	'namedGroups': true,
+	'lookbehind': true
+};
+
 const config = {
-	'ignoreCase': false,
-	'unicode': false,
-	'dotAll': false,
-	'useDotAllFlag': false,
-	'useUnicodeFlag': false,
-	'unicodePropertyEscape': false,
-	'namedGroup': false
+	'flags': {
+		'ignoreCase': false,
+		'unicode': false,
+		'dotAll': false,
+	},
+	'transform': {
+		'dotAllFlag': false,
+		'unicodeFlag': false,
+		'unicodePropertyEscapes': false,
+		'namedGroups': false,
+	},
+	get useUnicodeFlag() {
+		return this.flags.unicode && !this.transform.unicodeFlag;
+	}
 };
-const rewritePattern = (pattern, flags, options) => {
-	config.unicode = flags && flags.includes('u');
-	const regjsparserFeatures = {
-		'unicodePropertyEscape': config.unicode,
-		'namedGroups': true,
-		'lookbehind': options && options.lookbehind
-	};
-	config.ignoreCase = flags && flags.includes('i');
-	const supportDotAllFlag = options && options.dotAllFlag;
-	config.dotAll = supportDotAllFlag && flags && flags.includes('s');
-	config.namedGroup = options && options.namedGroup;
-	config.useDotAllFlag = options && options.useDotAllFlag;
-	config.useUnicodeFlag = options && options.useUnicodeFlag;
-	config.unicodePropertyEscape = options && options.unicodePropertyEscape;
-	if (supportDotAllFlag && config.useDotAllFlag) {
-		throw new Error('`useDotAllFlag` and `dotAllFlag` cannot both be true!');
+
+const validateOptions = (options) => {
+	if (!options) return;
+
+	for (const key of Object.keys(options)) {
```

Comment:
How about `for (const [key, value] of Object.entries(options))`?

Reply:
This package technically still supports Node.js 4. I wasn't going to propose to update it to Node.js 12 because Babel 7 still supports Node.js 6 too :sweat_smile: